sympy/stats/drv.py

```diff
@@ -100,16 +109,43 @@ def compute_characteristic_function(self, **kwargs):
         cf = summation(exp(I*t*x)*pdf, (x, self.set.inf, self.set.sup))
         return Lambda(t, cf)
 
+    def _characteristic_function(self, t):
+        return None
+
     def characteristic_function(self, t, **kwargs):
         """ Characteristic function """
+        if not kwargs:
+            cf = self._characteristic_function(t)
+            if cf is not None:
+                return cf
         return self.compute_characteristic_function(**kwargs)(t)
 
     def expectation(self, expr, var, evaluate=True, **kwargs):
         """ Expectation of expression over distribution """
         # TODO: support discrete sets with non integer stepsizes
+
         if evaluate:
-            return summation(expr * self.pdf(var),
-                         (var, self.set.inf, self.set.sup), **kwargs)
+            try:
+                # note: in order for this algorithm to be valid,
+                #   the characteristic function must have continuous
+                #   derivatives up to the highest power of the variable in the expression
+
+                t = Symbol('t', real=True, dummy=True)
```

Comment:
`dummy=True` will be ignored. This could be `Dummy('t', real=True)` (or `symbols('t', real=True, cls=Dummy)`).